test/UnitTests/MSTest.Analyzers.UnitTests/DynamicDataShouldBeValidAnalyzerTests.cs

```diff
@@ -1618,6 +1618,27 @@ public void TestMethod2(char c)
                 private static object GetData() => new List<object> { 1, 2, 3 };
             }
             
+            """,
+            VerifyCS.Diagnostic(DynamicDataShouldBeValidAnalyzer.MemberTypeRule).WithLocation(0).WithArguments("MyTestClass", "GetData"));
+
+    [TestMethod]
+    public async Task WhenDataIsPointer_Diagnostic()
+        => await VerifyCS.VerifyAnalyzerAsync(
+            """            
+            using System;
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            [TestClass]
+            public class MyTestClass
+            {
+                [TestMethod]
+                [{|#0:DynamicData(nameof(GetData), DynamicDataSourceType.Method)|}]
+                public void TestMethod2(char c)
+                {
+                }
+
+                private static IntPtr GetData() => new IntPtr(1);
```

Comment:
I think this is considered as `INamedTypeSymbol` and is not testing the proper check?

In that case, the following should test the case for getting `IPointerTypeSymbol`:

```csharp
private static unsafe int* GetData() => null;
```